lib/Wikia/src/Service/Helios/HeliosClientImpl.php

```diff
@@ -206,39 +204,63 @@ public function invalidateToken( $token, $userId )
 	 *
 	 * @return array - JSON string deserialized into an associative array
 	 */
-	public function generateToken( $userId )
-	{
+	public function generateToken( $userId ) {
 		return $this->request(
-			sprintf('users/%s/tokens', $userId),
+			sprintf( 'users/%s/tokens', $userId ),
 			[],
 			[],
 			[ 'method' => 'POST' ]
 		);
 	}
 
 	/**
-	* A shortcut method for register requests.
-	*/
-	public function register( $username, $password, $email, $birthdate, $langCode )
-	{
-			// Convert the array to URL-encoded query string, so the Content-Type
-			// for the POST request is application/x-www-form-urlencoded.
-			// It would be multipart/form-data which is not supported
-			// by the Helios service.
-			$postData = http_build_query( [
-				'username'  => $username,
-				'password'  => $password,
-				'email'     => $email,
-				'birthdate' => $birthdate,
-				'langCode'  => $langCode,
-			] );
-
-			return $this->request(
-				'users',
-				[],
-				$postData,
-				[ 'method'	=> 'POST' ]
-			);
+	 * A shortcut method for register requests.
+	 */
+	public function register( $username, $password, $email, $birthdate, $langCode ) {
+		// Convert the array to URL-encoded query string, so the Content-Type
+		// for the POST request is application/x-www-form-urlencoded.
+		// It would be multipart/form-data which is not supported
+		// by the Helios service.
+		$postData = http_build_query( [
+			'username'  => $username,
+			'password'  => $password,
+			'email'     => $email,
+			'birthdate' => $birthdate,
+			'langCode'  => $langCode,
+		] );
+
+		return $this->request(
+			'users',
+			[],
+			$postData,
+			[ 'method' => 'POST' ]
+		);
+	}
+
+	public function setPassword( $userId, $password ) {
+		$postData = http_build_query( [
+			'password' => $password,
+		] );
+
+		return $this->request(
+			"users/$userId/password",
+			[],
+			$postData,
+			[ 'method' => 'PUT', 'headers' => [ 'X-Wikia-UserId' => $userId ] ]
+		);
 	}
 
+	public function validatePassword( $password, $name ) {
+		$postData = http_build_query( [
+			'password' => $password,
+			'username' => $name,
+		] );
+
+		return $this->request(
+			"password/validation",
+			[],
+			$postData,
+			[ 'method' => 'POST' ]
```

Comment:
In general POST should be used for updating an entity. As validation doesn't have any side-effects, shouldn't we use GET here?